web_external/IndexView/index.jsx

```diff
@@ -1,21 +1,100 @@
 import React, { Component } from 'react';
+import events from 'girder/events';
+
 import TreeView from '../TreeView';
 import TrackAttribute from '../TrackAttribute';
+import annotationGeometryParser from '../util/annotationGeometryParser';
+import annotationActivityParser from '../util/annotationActivityParser';
+import annotationTrackParser from '../util/annotationTrackParser';
+import { restRequest } from 'girder/rest';
 import Viewer from '../Viewer';
 
 import './style.styl';
 
 class IndexView extends Component {
     constructor(props) {
         super(props);
+        this.toggleActivity = this.toggleActivity.bind(this);
+        this.toggleTrack = this.toggleTrack.bind(this);
+        this.state = {
+            annotationActivityContainer: null,
+            annotationTrackContainer: null,
+            annotationGeometryContainer: null
+        }
+    }
+    componentDidMount() {
+        // the states would be lift to redux store
+        events.on('v:item_selected', (itemModel) => {
+            this.setState({ itemModel });
+            downloadItemByName(itemModel.get('folderId'), 'activities.yml')
+                .then((raw) => {
+                    var annotationActivityContainer = annotationActivityParser(raw);
+                    this.setState({ annotationActivityContainer });
+                });
+            downloadItemByName(itemModel.get('folderId'), 'types.yml')
+                .then((raw) => {
+                    var annotationTrackContainer = annotationTrackParser(raw);
+                    this.setState({ annotationTrackContainer });
+                });
+            downloadItemByName(itemModel.get('folderId'), 'geom.yml')
+                .then((raw) => {
+                    var annotationGeometryContainer = annotationGeometryParser(raw);
+                    this.setState({ annotationGeometryContainer });
+                }).catch(() => {
+                    this.setState({ annotationGeometryContainer: false })
+                    events.trigger('g:alert', {
+                        icon: 'ok',
+                        text: 'Didn\'t find annotation files',
```

Comment:
This is going to bug me and keep bugging me... how about "Annotation files were not found" or "Failed to load annotation files"?

Reply:
+1 for "Annotation files were not found"